### PR TITLE
rating/age check

### PR DESCRIPTION
**Pull request by adumortier**
- added the checks on the `rating`
- added the checks on the `age`
- changed the `pet` `show`/`index` page to remove the `delete` link when a pet has been applied for
- added a helper method and associated test to check if the `rating` entered for a `review` is valid

### CODE REVIEW

File: app/controllers/pets_controller.rb [hunk 1]
```diff
@@ -14,7 +14,10 @@ def create
     if pet.valid? && pet.age >= 0
       redirect_to "/shelters/#{shelter.id}/pets"
     else
-      (flash[:age_error] = "You must enter a positive value for the age") if (pet.age.to_i < 0)
+      if (pet.age.to_i < 0)
```
Comment (adumortier) [inline app/controllers/pets_controller.rb]: negative `age` triggers a flash message.`create` method needs unloading

File: app/controllers/pets_controller.rb [hunk 2]
```diff
@@ -31,10 +34,14 @@ def edit
   def update
     shelter = Shelter.find(params[:shelter_id])
     pet = Pet.find(params[:pet_id])
-    if pet.update(pet_params) && (pet_params[:age].to_i >= 0)
+    if (pet_params[:age].to_i < 0)
```
Comment (adumortier) [inline app/controllers/pets_controller.rb]: negative `age` triggers flash message and prevents the update of the `pet`

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -5,15 +5,20 @@ def new
...
   def create
+    score = review_params[:rating]
     shelter = Shelter.find(params[:shelter_id])
     new_review = shelter.reviews.new(review_params)#######changed from create to new
-    if new_review.save
+    if new_review.save && Review.valid_rating?(score)  
       flash[:happy] = 'Review successfully submitted!'
       redirect_to "/shelters/#{shelter.id}"
     else
+      if !Review.valid_rating?(score)
```
Comment (adumortier) [inline app/controllers/reviews_controller.rb]: checks that `rating` entered is valid before creating the `review`

File: app/models/review.rb [hunk 4]
```diff
@@ -2,4 +2,8 @@ class Review < ApplicationRecord
   validates_presence_of :title, :rating, :content, :image
 
   belongs_to :shelter
+
+  def self.valid_rating?(score)
```
Comment (adumortier) [inline app/models/review.rb]: this helper method checks if a rating is between 1 and 5

File: spec/features/applications/new_application_spec.rb [hunk 5]
```diff
@@ -40,14 +40,15 @@
...
-    within("span#pet_#{@pet_1.id}") do
-      check "pet_ids_"
-    end
-
+    
   end
   
   it 'you can apply for a pet that will then remove that pet from the favorites page' do
     
+    within("span#pet_#{@pet_1.id}") do
```
Comment (adumortier) [inline spec/features/applications/new_application_spec.rb]: copied the selection of the first `pet` to each of the tests to allow the last test to have no `pet` selected

File: spec/features/applications/new_application_spec.rb [hunk 6]
```diff
@@ -114,6 +124,23 @@
     end
 
   end
+
+  it "shows a flash message if not pets where selected for an application" do
```
Comment (adumortier) [inline spec/features/applications/new_application_spec.rb]: tests that submitting an application without pet triggers a flash message

File: spec/features/pets/new_spec.rb [hunk 7]
```diff
@@ -46,5 +46,25 @@
...
+
+    fill_in :image, with: 'https://s3.amazonaws.com/cdn-origin-etr.akc.org/wp-content/uploads/2017/11/12234558/Chinook-On-White-03.jpg'
+    fill_in :name, with: "Cracky"
+    fill_in :description, with: "smart"
+    fill_in :age, with: "-3"
+    select 'F', :from => :sex
+
+    click_button "Create Pet"
+    expect(page).to have_content("You must enter a positive value for the age")
+    expect(current_path).to eq("/shelters/#{@shelter.id}/pets/new")
```
Comment (agallant121) [inline spec/features/pets/new_spec.rb]: Expect that negative ages are not saved into the db

**Pull request merged**